src/db/mock.rs

```diff
@@ -0,0 +1,34 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Mock db implementation with methods stubbed to return default values.
+
+use futures::future;
+
+use super::*;
+
+#[derive(Debug)]
+pub struct MockDb;
+
+macro_rules! mock_db_method {
+    ($name:ident, $type:ident) => {
+        fn $name(&self, _params: params::$type) -> Box<Future<Item = results::$type, Error = DbError>> {
+            Box::new(future::result(Ok(results::$type::default())))
+        }
+    }
+}
+
+impl Db for MockDb {
+    mock_db_method!(get_collections, GetCollections);
```

Comment:
I'm a bit wary of macro's for trait definitions as they impair the readability of the trait with fairly minimal gain.

Reply:
Understood. Should I get rid of the `db::params` and `db::results` abstractions too in that case, because the the method signatures aren't actually that useful otherwise, all the meaningful definitions are tied up in those two modules?